### DISCUSSION TITLE
Feature request: 'Espanso edit' through UI

### DISCUSSION THREAD

**Discussion by kars-h** [espanso/espanso#1234] [Feature requests and Ideas]
Loving the new UI in the latest 2.x alpha. 
To make it complete, it makes a lot of sense to me to have 2 ways through the UI to edit the config file.
1. In the taskbar: 
![image](https://user-images.githubusercontent.com/70519718/138108887-d19898e7-86eb-41f7-8221-14f7fd5f7e43.png)

2. In the search bar, perhaps even by just typing 'edit' 
![image](https://user-images.githubusercontent.com/70519718/138108964-b25ba7f1-0b62-4323-9d10-2f28cb990f01.png)

(I'm on W10 btw).

**Comment by federico-terzi** [comment]
Hey @kars-h,

Thank you for the feedback! This is something that will surely be implemented in the future, but I first want to build the general "GUI editor", which will be the next major milestone in 2022.

Until then, stay tuned :)

**Comment by jczanfona** [comment]
It is certainly a necessary thing. I propose two places:
- In the icon tray, by right clicking on the icon
- In the search bar when clicking on the icon (although I think the most appropriate thing is that this icon links to the web)

We still waiting. Cheer up with everything, it is looking VERY GOOD.
The wait is worth it.

**Comment by falschparker82** [comment]
Hi Federico,

thanks for your efforts... just signed up our org as a proud supporter.

I'd like to chime in with this feature, now that I don't use the terminal much anymore for espanso, this would be the last great addition to the UI dropdown (and/or as the edit shortcut proposed, which I'd like just as much).

As for GUI editor: I don't know what you've planned there, but "just" launching the system editor bound to .yml files (Sublime in my case which I love & trust) would be an absolute bless already and I'd definitely take that over a custom editor. As most of us nerds are pretty particular about our editors and key configuration and your target group includes hardcore devs, I'd wager I'm not the only one there.

Keep up the great work and happy holidays!

**Comment by federico-terzi** [comment]
Hey @falschparker82,

Thanks for the support! I really appreciate it :)

> As for GUI editor: I don't know what you've planned there, but "just" launching the system editor bound to .yml files (Sublime in my case which I love & trust) would be an absolute bless already and I'd definitely take that over a custom editor. As most of us nerds are pretty particular about our editors and key configuration and your target group includes hardcore devs, I'd wager I'm not the only one there.

I'm still investigating the optimal way to bring the editor to life, though it will take a while :) In the meanwhile, launching the system editor through the tray menu seems like an easy solution already! I'll investigate

Cheers :)

**Comment by kpym** [comment]
May be `default.yaml` can be used to configure the GUI menus, something like this:
```yaml
menu:
  - text: Edit config
    command: code '%CONFIG%/match/'
```
